File: src/test/java/no/novari/flyt/history/repository/utils/performance/PageSizePerformanceTestCase.java
package no.novari.flyt.history.repository.utils.performance;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.kafka.support.JavaUtils;

import java.time.Duration;
import java.util.StringJoiner;

@Getter
@AllArgsConstructor
public class PageSizePerformanceTestCase {
    private final int requestedMaxSize;
    private final int expectedSize;
    private final Duration maxElapsedTime;

    @Override
    public String toString() {
        StringJoiner joiner = new StringJoiner(", ", "(", ")");
        JavaUtils.INSTANCE.acceptIfNotNull(
                requestedMaxSize, requestedSize -> joiner.add("requestedSize=" + requestedSize)
        );
        JavaUtils.INSTANCE.acceptIfNotNull(
                expectedSize, expectedSize -> joiner.add("expectedSize=" + expectedSize)
        );
        JavaUtils.INSTANCE.acceptIfNotNull(
                maxElapsedTime, maxElapsedTime -> joiner.add("maxElapsedTime=" + maxElapsedTime)
        );
        return joiner.toString();
    }

}

File: src/test/kotlin/no/novari/flyt/history/repository/utils/performance/PageSizePerformanceTestCase.kt
package no.novari.flyt.history.repository.utils.performance

import java.time.Duration
import java.util.StringJoiner

class PageSizePerformanceTestCase(
    val requestedMaxSize: Int,
    val expectedSize: Int,
    val maxElapsedTime: Duration,
) {
    override fun toString(): String {
        val joiner = StringJoiner(", ", "(", ")")
        joiner.add("requestedSize=$requestedMaxSize")
        joiner.add("expectedSize=$expectedSize")
        joiner.add("maxElapsedTime=$maxElapsedTime")
        return joiner.toString()
    }
}
